controller/src/main/java/io/pravega/controller/store/client/StoreClientFactory.java

```diff
@@ -94,11 +102,14 @@ static CuratorFramework createZKClient(ZKClientConfig zkClientConfig, Supplier<B
     }
 
     @VisibleForTesting
-    static class ZKClientFactory implements ZookeeperFactory {
+    public static class ZKClientFactory implements ZookeeperFactory {
```

Comment:
We should be able to test this class without making it public. The scope of package protected should be sufficient to test it. Let's rework the test cases so that we don't have to do it.

Reply:
At the moment, I could restrict the visibility of this nested class to `protected`, so a mock class for test can extend its functionality and close the ZK client when desired in a test. Let's see if this solution is better than the previous one.